src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java

```diff
@@ -274,6 +274,8 @@ private void adjustProjectName(Map<String, String> replacements, String projectK
      */
     public ProgrammingExercise importProgrammingExercise(ProgrammingExercise originalProgrammingExercise, ProgrammingExercise newExercise, boolean updateTemplate,
             boolean recreateBuildPlans) {
+        // remove all non-alphanumeric characters from the short name. This gets already done in the client, but we do it again here to be sure
+        newExercise.setShortName(newExercise.getShortName().replaceAll("[^a-zA-Z0-9]", ""));
```

Comment:
Can we reuse / move this into a utility method in StringUtils (e.g. escapeForShortName)?

Reply:
Where does it needs to be reused? I don't see the benefit